Frends.AzureBlobStorage.UploadBlob/Frends.AzureBlobStorage.UploadBlob/UploadBlob.cs

```diff
@@ -348,17 +479,33 @@ private static Encoding GetEncoding(string target)
         };
     }
 
-    private static void CheckParameters(Destination destination, FileInfo fi)
+    private static void CheckParameters(Destination destination, Source source)
     {
+        if (!string.IsNullOrWhiteSpace(source.SourceDirectory) && source.SourceType is UploadSourceType.Directory && !Directory.Exists(source.SourceDirectory))
+            throw new Exception(@$"Source directory {source.SourceDirectory} doesn't exists.");
+        if (!string.IsNullOrWhiteSpace(source.SourceDirectory) && source.SourceType is UploadSourceType.Directory && !Directory.EnumerateFileSystemEntries(source.SourceDirectory).Any())
+            throw new Exception(@$"Source directory {source.SourceDirectory} is empty.");
+        if (!string.IsNullOrWhiteSpace(source.SourceFile) && source.SourceType is UploadSourceType.Directory)
+            throw new Exception("Source.SourceFile must be empty when Source.SourceType is Directory.");
+        if (string.IsNullOrWhiteSpace(source.SourceDirectory) && source.SourceType is UploadSourceType.Directory)
+            throw new Exception("Source.SourceDirectory value is empty.");
+
+        if (!string.IsNullOrWhiteSpace(source.SourceFile) && source.SourceType is UploadSourceType.Directory && File.Exists(source.SourceFile))
+            throw new Exception(@$"Source file {source.SourceFile} doesn't exists.");
+        if (!string.IsNullOrWhiteSpace(source.SourceDirectory) && source.SourceType is UploadSourceType.File)
+            throw new Exception("Source.SourceDirectory must be empty when Source.SourceType is Directory.");
+        if (string.IsNullOrWhiteSpace(source.SourceFile) && source.SourceType is UploadSourceType.File)
+            throw new Exception("Source.SourceFile not found.");
+        if (string.IsNullOrWhiteSpace(source.SourceFile) && source.SourceType is UploadSourceType.File)
+            throw new Exception("Source.SourceFile not found.");
+
+
         if (destination.ConnectionMethod is ConnectionMethod.OAuth2 && (destination.ApplicationID is null || destination.ClientSecret is null || destination.TenantID is null || destination.StorageAccountName is null))
             throw new Exception("Destination.StorageAccountName, Destination.ClientSecret, Destination.ApplicationID and Destination.TenantID parameters can't be empty when Destination.ConnectionMethod = OAuth.");
         if (string.IsNullOrWhiteSpace(destination.ConnectionString) && destination.ConnectionMethod is ConnectionMethod.ConnectionString)
             throw new Exception("Destination.ConnectionString parameter can't be empty when Destination.ConnectionMethod = ConnectionString.");
         if (string.IsNullOrWhiteSpace(destination.ContainerName))
             throw new Exception("Destination.ContainerName parameter can't be empty.");
-
-        if (!fi.Exists)
-            throw new Exception("Source file doesn't exists.");
     }
 
     private static void OnPluginUnloadingRequested(AssemblyLoadContext obj)

```

Comment:
You can remove this method because it didn't have the wanted effect on memory leak issue.